File: VB/Controls/LogViewerVB.WinForms/Extensions/ServicesExtension.vb
Imports System.Runtime.CompilerServices
Imports LogViewerVB.Core
Imports LogViewerVB.Core.ViewModels
Imports Microsoft.Extensions.DependencyInjection
Imports Microsoft.Extensions.Hosting

Public Module ServicesExtension

    <Extension>
    Public Function AddLogViewer(builder As HostApplicationBuilder) As HostApplicationBuilder

        builder.Services.AddSingleton(Of ILogDataStore, Logging.LogDataStore)
        builder.Services.AddSingleton(Of LogViewerControlViewModel)
        builder.Services.AddTransient(Of LogViewerControl)

        Return builder

    End Function

End Module

File: CSharp/Controls/LogViewer.WinForms/Extensions/ServicesExtension.cs
using LogViewer.Core;
using LogViewer.Core.ViewModels;
using Microsoft.Extensions.DependencyInjection;
using Microsoft.Extensions.Hosting;

namespace LogViewer.WinForms;

public static class ServicesExtension
{
    public static HostApplicationBuilder AddLogViewer(this HostApplicationBuilder builder)
    {
        builder.Services.AddSingleton<ILogDataStore, Logging.LogDataStore>();
        builder.Services.AddSingleton<LogViewerControlViewModel>();
        builder.Services.AddTransient<LogViewerControl>();

        return builder;
    }
}